src/dlaf_fortran.f90

```diff
@@ -239,6 +246,42 @@ end subroutine dlaf_pssyevd_c
 
    end subroutine dlaf_pssyevd
 
+   subroutine dlaf_pssyevd_partial_spectrum(uplo, n, a, ia, ja, desca, w, z, iz, jz, descz, il, iu, info)
+      character, intent(in) :: uplo
+      integer, intent(in) :: n, ia, ja, iz, jz
+      integer(kind=i8), intent(in) :: il, iu
+      integer, dimension(9), intent(in) :: desca, descz
+      integer, target, intent(out) :: info
+      real(kind=sp), dimension(:, :), target, intent(inout) :: a, z
+      real(kind=sp), dimension(:), target, intent(out) :: w
+
+      interface
+         subroutine dlaf_pssyevd_ps_c(uplo_, n_, a_, ia_, ja_, desca_, w_, z_, iz_, jz_, descz_, il_, iu_, info_) &
+            bind(C, name='dlaf_pssyevd_partial_spectrum')
+
+            import :: c_int, c_ptr, c_signed_char, c_ptrdiff_t
+
+            integer(kind=c_signed_char), value :: uplo_
+            integer(kind=c_int), value :: n_, ia_, ja_, iz_, jz_
+            integer(kind=c_ptrdiff_t), value :: il_, iu_
+            type(c_ptr), value :: a_, w_, z_
+            integer(kind=c_int), dimension(9) :: desca_, descz_
+            type(c_ptr), value :: info_
+         end subroutine dlaf_pssyevd_ps_c
+      end interface
+
+      info = -1
+
+      call dlaf_pssyevd_ps_c(iachar(uplo, c_signed_char), n, &
+                             c_loc(a(1, 1)), ia, ja, desca, &
+                             c_loc(w(1)), &
+                             c_loc(z(1, 1)), iz, jz, descz, &
+                             il - 1, iu, &
```

Comment:
Just as a reminder for me: The Fortran API uses 1-based indexing, which is why there's a `-1` here? And the "upper" (the `u` in `iu` is upper?) is inclusive unlike the C API where it's exclusive?

Reply:
Yes, the Fortran API follows the ScaLAPACK convention: 1-based indexing (`IL >= 1`) and the last index is inclusive (`min(IL,N) <= IU <= N`). For the C++/C API we have 0-based indexing and the last index is exclusive. `il` and `iu` are also from ScaLAPACK and indeed stand for "index lower" and "index upper".

Reply:
Wait:
This is a major issue we didn't discuss in DLAF and was not noticed.
ScaLAPACK C-API Should use base 0 or base 1 index?
If we want to claim ScaLAPACK drop-in it should be base-1 (normal C-API is for who wants 0 base indices).

What do you think?
We are still in time for a 0.7.2 patch release for DLAF and deprecation of 0.7.[0-1]

What do you think? @RMeli @msimberg 

Note: index convertion should happen here IMO.
https://github.com/eth-cscs/DLA-Future/blob/master/src/c_api/eigensolver/eigensolver.h#L93-L94

Reply:
Ah! Good point. I didn't think this through; we already impose this constraint in the sub-matrices indices (`ai`, `aj`, ...). I don't particularly like using base 1 indexing in C, but I just checked `lapackpp` and they indeed use base 1 indexing, so probably we should do too.